dags/sustainment/information_url_checker.py

```diff
@@ -0,0 +1,88 @@
+import requests
+import logging
+from datetime import datetime, timedelta
+
+from airflow import DAG
+from airflow.models import Variable
+from airflow.operators.python import PythonOperator
+from ckan_operators.package_operator import GetCkanPackageListOperator
+from utils_operators.slack_operators import GenericSlackOperator, task_failure_slack_alert
+
+# init hardcoded vars for these dags
+ACTIVE_ENV = Variable.get("active_env")
+CKAN_CREDS = Variable.get("ckan_credentials_secret", deserialize_json=True)
+CKAN = CKAN_CREDS[ACTIVE_ENV]["address"]
+CKAN_APIKEY = CKAN_CREDS[ACTIVE_ENV]["apikey"]
+
+with DAG(
+    'information-url-checker',
+    default_args={
+                "owner": "Yanan",
+                "depends_on_past": False,
+                "email": "yanan.zhang@toronto.ca",
+                "email_on_failure": False,
+                "email_on_retry": False,
+                "retries": 1,
+                "retry_delay": 3,
+                "on_failure_callback": task_failure_slack_alert,
+                "tags": ["sustainment"]
+            },
+    description='Check if information_url(Under dataset description section) accessible, send notification to slack on failure urls',
+    schedule_interval="@once",
+    start_date=datetime(2023, 1, 27, 0, 0, 0),
+) as dag:
+    
+    def information_url_checker(**kwargs):
+        ti = kwargs.pop("ti")
+        package_list = ti.xcom_pull(task_ids="get_package_list")
+        address = kwargs.pop("address")
+        
+        result_info = {}
+        for package_name in package_list:
+            url = address + "api/3/action/package_show?id=" + package_name
+            response = requests.get(url).json()['result']
+            if "information_url" in response.keys():
+                information_url = response['information_url']
+                if 'http' in information_url:
+                    call = information_url
+                else:
+                    call = 'https://' + information_url
+                logging.info(f"{call}")
+                try:
+                    response = requests.get(call)
+                    status = response.status_code
+                    reason = response.reason
+                    logging.info(f"---Status-{status}---Reason--{reason}---{package_name}----{call}----------")
+                    if status == 404:
```

Comment:
404, I think, is one of many problematic `status_code` values we can get.

I think anything that starts with 400 or 500 is a problem :/ Basically, if its not 2xx or 3xx then something is wrong.

Also! 
If a response code is 302, it means the URL is being redirected to another site. That's bad if it's being redirected to the toronto.ca homepage. I think we should add logic that says something like:

```
if status = 302:
    if new_destination == "toronto.ca":
        <mark as failed>

Reply:
I like this idea! although currently our response code is either 200 or 404, but it could happen and it's great to bring to our attention! will add more condition check based on different code: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status